envision/server.py

```diff
@@ -231,14 +234,24 @@ async def run_loop():
                         frame_ptr = self._frames.start_frame
                     frames_to_send = [frame_ptr]
                     self._seek = None
-
-                assert len(frames_to_send) > 0
-                closed = self._push_frames_to_web_client(frames_to_send)
-                if closed:
-                    self._log.debug("Socket closed, exiting")
-                    return
-
-                frame_ptr, frames_to_send = self._wait_for_next_frame(frame_ptr)
+                else:
+                    # If not seeking, try to get next frames normally
+                    delay = self._calculate_frame_delay(frame_ptr)
+                    time.sleep(delay)
+                    frames_to_send = []
+                    while (
+                        frame_ptr.next_
+                        and len(frames_to_send) <= self._message_frame_volume
+                    ):
+                        frame_ptr = frame_ptr.next_
+                        frames_to_send.append(frame_ptr)
```

Comment:
Is there a reason why this should not always happen? (assuming `frames_to_send` is instead reset at the beginning of every loop)

Reply:
I refactored the loop to take out this if/else branch. I also moved the sleep to happen if there are no new frames. This seems to work well from testing (I also experimented with removing the sleep completely, which appeared to break some things).